Terminal.Gui/View/ViewText.cs

```diff
@@ -69,23 +69,22 @@ public virtual string Text
                 Id = _text;
             }
 #endif
-            OnTextChanged (new CancelEventArgs<string> (old, Text));
+            OnTextChanged ();
         }
     }
 
     /// <summary>
     /// Called when the <see cref="Text"/> has changed. Fires the <see cref="TextChanged"/> event.
     /// </summary>
-    /// <param name="args">The event arguments</param>
-    public void OnTextChanged (CancelEventArgs<string> args)
+    public void OnTextChanged ()
     {
-        TextChanged?.Invoke (this, args);
+        TextChanged?.Invoke (this, EventArgs.Empty);
     }
 
     /// <summary>
     ///     Text changed event, raised when the text has changed.
     /// </summary>
-    public event EventHandler<CancelEventArgs<string>> TextChanged;
+    public event EventHandler<EventArgs> TextChanged;
```

Comment:
Did you mean to use our `EventArgs<T>` you created?

If not, then the event can just be `public event EventHandler? TextChanged;`[^1]. That's actually exactly how that same event is defined in windows forms. Anything derived from EventArgs is therefore accepted.

WPF defines a much richer event [delegate](https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.textchangedeventhandler?view=windowsdesktop-8.0) and [eventargs type](https://learn.microsoft.com/en-us/dotnet/api/system.windows.controls.textchangedeventargs?view=windowsdesktop-8.0) for it, which includes some goodies that make undo handling easier. But that's also got stuff not relevant to TG. Just another reference.


[^1]: The EventHandler delegate [already uses EventArgs for e](https://github.com/dotnet/runtime/blob/v8.0.6/src/libraries/System.Private.CoreLib/src/System/EventHandler.cs#L6). The generic is fine, but will only make a difference if T of `EventArgs<T>` is a value type, since reference types share the runtime concrete implementation.

Reply:
Yep. already fixed.

Reply:
No. I changed the semantics of the TextChanged event.

Previously it conveyed old/new but was (clearly) not cancelable. Earlier in this PR I (stupidly) made it `EventHandler<CancelEventArgs<string>>` to make it convey old/new as it had before. By using `CancelEventArgs` I made it confusing.

`Text` can be HUGE. I was an error to originally think it was ok for `TextChanged` to convey EITHER the old or new text as there's no simple way to avoid copies of the string. And some subs may not even care about the new text, much less the old. If a sub wants the new text they can simply access `((View)sender).Text`.  And if a sub wants the old text, they should be using `TextChanging`.

Reply:
> public event EventHandler? TextChanged;

Done.